apisix/discovery/consul/init.lua

```diff
@@ -197,21 +197,22 @@ local function get_opts(consul_server, is_catalog)
         port = consul_server.port,
         connect_timeout = consul_server.connect_timeout,
         read_timeout = consul_server.read_timeout,
+        default_args = {
+            token = consul_server.token,
+        }
     }
     if not consul_server.keepalive then
         return opts
     end
 
     if is_catalog then
-        opts.default_args = {
-            wait = consul_server.wait_timeout, --blocked wait!=0; unblocked by wait=0
-            index = consul_server.catalog_index,
-        }
+        opts.default_args.wait = consul_server.wait_timeout --blocked wait!=0; unblocked by wait=0
+        opts.default_args.index = consul_server.catalog_index
+        opts.default_args.token = consul_server.token
```

Comment:
Why do you add `opts.default_args.token = consul_server.token` again, you have done this at line 200 ~ 202.

Reply:
Yeah, that was a mistake sorry. I've fixed it.